go/vt/vttablet/tabletmanager/rpc_actions.go

```diff
@@ -142,9 +142,12 @@ func (tm *TabletManager) RunHealthCheck(ctx context.Context) {
 	tm.QueryServiceControl.BroadcastHealth()
 }
 
-func convertBoolToSemiSyncAction(semiSync bool) SemiSyncAction {
+func (tm *TabletManager) convertBoolToSemiSyncAction(semiSync bool) SemiSyncAction {
 	if semiSync {
 		return SemiSyncActionSet
 	}
-	return SemiSyncActionUnset
+	if tm.MysqlDaemon.SemiSyncExtensionLoaded() {
+		return SemiSyncActionUnset
+	}
```

Comment:
Depends on what we want the result to be if `semiSync == true && tm.MysqlDaemon.SemiSyncExtensionLoaded() == false`.

With the current code it'd return `SemiSyncActionSet` and likely cause an error to be thrown as the code would likely try to run some semi-sync related queries. This would help us know when there is a bug where `semiSync` is true when it really shouldn't be. With your suggestion, we'd just skip those steps, and that bug would likely be unnoticed in this instance.

One potential way for this to occur is say a user set durability policy to semi-sync, but didn't load the proper plugins in their mysql. In that case, the current code would try to set semi-sync settings, and fail with a loud error message informing the user that they're missing the right plugins. (Not sure if this would actually occur, but just a hypothetical)